### PR TITLE
[SYCL] Make full copy of buffer from device to host and back.

### PR DESCRIPTION
**Pull request by KarachunIvan**
Case when accessor has a smaller range than buffer causes lost
of initial data after copying data back to host.
The fix is to always copy whole buffer.

Signed-off-by: Ivan Karachun <ivan.karachun@intel.com>

### CODE REVIEW

File: sycl/source/detail/memory_manager.cpp [hunk 1]
```diff
@@ -306,7 +306,7 @@ void MemoryManager::fill(SYCLMemObjT *SYCLMemObj, void *Mem, QueueImplPtr Queue,
 
   if (Dim == 1) {
     PI_CALL(RT::piEnqueueMemFill(
-        Queue->getHandleRef(), pi::pi_cast<RT::PiMem>(Mem), Pattern, PatternSize, Offset[0],
+        Queue->getHandleRef(), pi::pi_cast<RT::PiMem>(Mem), Pattern, PatternSize, Offset[0] * ElementSize,
```
Comment (romanovvlad) [inline sycl/source/detail/memory_manager.cpp]: It seems this change is not related to buffers copying, please move to separate commit.

File: sycl/test/basic_tests/buffer/buffer_full_copy.cpp [hunk 2]
```diff
@@ -0,0 +1,183 @@
+// RUN: %clang -std=c++11 %s -o %t1.out -lstdc++ -lOpenCL -lsycl
```
Comment (bader) [inline sycl/test/basic_tests/buffer/buffer_full_copy.cpp]:
```suggestion
// RUN: %clangxx %s -o %t1.out -lOpenCL
```

File: sycl/test/basic_tests/buffer/buffer_full_copy.cpp [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+// RUN: env SYCL_DEVICE_TYPE=HOST %t1.out
+// RUN: %clang -std=c++11 -fsycl %s -o %t2.out -lstdc++ -lOpenCL -lsycl
```
Comment (bader) [inline sycl/test/basic_tests/buffer/buffer_full_copy.cpp]:
```suggestion
// RUN: %clangxx -fsycl %s -o %t2.out -lOpenCL
```

Comment (KarachunIvan) [reply sycl/test/basic_tests/buffer/buffer_full_copy.cpp]: Done

File: sycl/test/basic_tests/buffer/buffer_full_copy.cpp (hunk 2 above)
Comment (KarachunIvan) [reply sycl/test/basic_tests/buffer/buffer_full_copy.cpp]: Done

File: sycl/test/basic_tests/buffer/buffer_full_copy.cpp [hunk 4]
```diff
@@ -0,0 +1,183 @@
+// RUN: %clangxx %s -o %t1.out -lstdc++ -lOpenCL -lsycl
```
Comment (bader) [inline sycl/test/basic_tests/buffer/buffer_full_copy.cpp]: Please, remove linking with SYCL and stdc++ libraries.

Comment (KarachunIvan) [reply sycl/test/basic_tests/buffer/buffer_full_copy.cpp]: Done
